src/tests/roc_audio/test_profiler.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/profiler.h"
+#include "roc_core/heap_allocator.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+const core::nanoseconds_t interval = 1000;
+const int sample_rate = 300;
+const int num_channels = 1;
+core::HeapAllocator allocator;
+
+} // namespace
+
+TEST_GROUP(profiler) {};
```

Comment:
Not sure how to initialize test group properly. Any advice would be appreciated.

Reply:
Test group is actually a class. If you need any setup/teardown procedure common to all tests, you can add it here. If you don't need one, you can leave it empty.

https://cpputest.github.io/manual.html#setup_teardown

Reply:
Is there a way to inherit this class from Profiler like you can in Gtest test classes. This would allow me to better test the private member variables etc. and would make my test significantly cleaner and better.

Reply:
I think no.

Also, in Roc we try to follow the known practice of testing public methods only (blackbox testing) and designing class interfaces with testability in mind. This approach forces you to think more on decomposition and hopefully leads to a better division of responsibilities between classes.

My rule of thumb is: if you can't test the class as a blackbox, likely it currently lies on a wrong abtraction level. You should either decrease it by extracting part of its functionality into a new class which you can test separately, or increase it by moving some functionality from higher levels into the class. There are exceptions, though.

What specific problems do you have with Profiler currently?

Reply:
No worries. Would've liked that feature just to ensure that all internal calculations are being properly calculated but I can manage the same outcome using abstraction.